src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -458,7 +458,7 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
   def compileSchemaDeclarations(stmts: List[SchemaDeclaration], ss: CompilationState): CompiledBlocks = {
     var schemas = new ListBuffer[String]()
     for (stmt <- stmts) {
-      if ((stmt.a.name startsWith "dd_new_") && (ss.inferenceRuleGroupByHead contains stmt.a.name)) {
+      if ((stmt.a.name startsWith "dd_new_") && (ss.isQueryTerm(stmt.a.name))) {
```

Comment:
Looks like here's another idiom that's worth extracting a function for: `isForIncremental(stmt: Statement): Boolean` (or a with clearer name).  I see it's repeated a lot of times below.